lib/segment/src/segment_constructor/rocksdb_builder.rs

```diff
@@ -0,0 +1,87 @@
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use parking_lot::{RwLock, RwLockReadGuard};
+
+use super::segment_constructor_base::get_vector_name_with_prefix;
+use crate::common::operation_error::{OperationError, OperationResult};
+use crate::common::rocksdb_wrapper;
+use crate::types::{SegmentConfig, SparseVectorStorageType};
+
```

Comment:
A comment like this might be helpful:

```suggestion
/// Struct to optionally create and open a RocksDB instance in a lazy way.
/// Used as helper to eventually completely remove RocksDB.
```